servers/zms/src/main/java/com/yahoo/athenz/zms/store/file/FileConnection.java

```diff
@@ -405,6 +405,47 @@ public boolean insertDomainTemplate(String domainName, String templateName, Stri
         return true;
     }
 
+    @Override
+    public boolean updateDomainTemplate(String domainName, String templateName, TemplateMetaData templateMetaData) {
+        DomainStruct domainStruct = getDomainStruct(domainName);
+        ArrayList<TemplateMetaData> templateMetalist = new ArrayList<>();
+        if (domainStruct == null) {
+            throw ZMSUtils.error(ResourceException.NOT_FOUND, "domain not found", "updateDomainTemplate");
+        }
+        if (domainStruct.getTemplates() == null) {
+            domainStruct.setTemplates(new ArrayList<>());
+        }
+        ArrayList<String> templates = domainStruct.getTemplates();
+        if (!templates.contains(templateName)) {
+            templates.add(templateName);
+        }
+
+        if (domainStruct.getTemplateMeta() == null) {
+            domainStruct.setTemplateMeta(new ArrayList<>());
+        }
+        ArrayList<TemplateMetaData> templateMetaList = domainStruct.getTemplateMeta();
+        if (!templateMetaList.isEmpty()) {
+            for (TemplateMetaData meta : templateMetaList) {
+                if (meta.getTemplateName().equals(templateName)) {
+                    meta.setCurrentVersion(templateMetaData.getLatestVersion());
+                }
+            }
+        } else {
+            TemplateMetaData templateMeta = new TemplateMetaData();
+            templateMeta.setTemplateName(templateName);
+            templateMeta.setKeywordsToReplace(templateMetaData.getKeywordsToReplace());
+            templateMeta.setAutoUpdate(templateMetaData.getAutoUpdate());
+            templateMeta.setDescription(templateMetaData.getDescription());
+            templateMeta.setTimestamp(templateMetaData.getTimestamp());
```

Comment:
these lines 436-439  are not necessary since the domain object only needs to contain the name of the template and the latest version and no other detail.

Reply:
Done.